module/os/freebsd/zfs/zfs_debug.c

```diff
@@ -218,12 +218,19 @@ __dprintf(boolean_t dprint, const char *file, const char *func,
 	}
 
 	/*
-	 * Get rid of trailing newline.
+	 * Get rid of trailing newline for dprintf logs.
 	 */
-	nl = strrchr(buf, '\n');
-	if (nl != NULL)
-		*nl = '\0';
+	if (dprint && buf[0] != '\0') {
+		nl = &buf[strlen(buf) - 1];
+		if (*nl == '\n')
+			*nl = '\0';
+	}
 
+	/*
+	 * To get this data:
+	 *
+	 * $ sysctl -n kstat.zfs.misc.dbgmsg
```

Comment:
This seems redundant with the comment on line 49, no?

Reply:
Linux has the same. More is not less. ;)  Or I can remove from both, if you insist.

Reply:
I mean, I added the previous comment, I am not trying to suggest removing both, just more trying to understand if you were suggesting adding it for just parity or because you didn't find the sole comment above discoverable enough.

One of my outstanding "TODO" projects is "make the FreeBSD-specific equivalents of things more discoverable" since basically all the documentation is specific to the Linux versions, and that's not the most useful thing even if 90% of the names are the same.